services/scabbard/src/cli/key.rs

```diff
@@ -0,0 +1,200 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::PathBuf;
+
+use sawtooth_sdk::signing::secp256k1::Secp256k1PrivateKey;
+
+use super::error::CliError;
+
+/// Load a private key from the local filesystem.
+///
+/// If the argument is a file path (contains a '/'), this will attempt to load the key file from
+/// the specified location. If the argument is not a file path, this will attempt to load the
+/// file from the $HOME/.splinter/keys directory; when loading from this directory, the '.prive'
+/// file extension is optional.
+pub fn load_signing_key(key: &str) -> Result<Secp256k1PrivateKey, CliError> {
+    let file_path = determine_key_file_path(key)?;
+
+    let key_file = File::open(file_path).map_err(|err| {
+        CliError::action_error_with_source("failed to open private key file", err.into())
+    })?;
+    let mut key_file_reader = BufReader::new(key_file);
+
+    let mut raw_key_string = String::new();
+    key_file_reader
+        .read_line(&mut raw_key_string)
+        .map_err(|err| {
+            CliError::action_error_with_source("failed to read private key file", err.into())
+        })?;
+
+    let key_string = raw_key_string.trim();
+    if key_string.is_empty() {
+        return Err(CliError::action_error("private key file is empty"));
+    }
+
+    let signing_key = Secp256k1PrivateKey::from_hex(key_string).map_err(|err| {
+        CliError::action_error_with_source("failed to read valid private key from file", err.into())
+    })?;
+
+    Ok(signing_key)
+}
+
+fn determine_key_file_path(key: &str) -> Result<PathBuf, CliError> {
+    if key.contains('/') {
+        Ok(PathBuf::from(key))
+    } else {
+        let mut path = dirs::home_dir().ok_or_else(|| {
+            CliError::action_error("failed to load signing key: unable to determine home directory")
+        })?;
+        path.push(".splinter");
+        path.push("keys");
+        path.push(key);
+
+        if path.exists() {
+            Ok(path)
+        } else {
+            path.set_extension("priv");
+            if path.exists() {
+                Ok(path)
+            } else {
+                Err(CliError::action_error(
+                    "failed to load signing key: could not be found",
+                ))
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    use std::fs::create_dir_all;
+    use std::io::Write;
+    use std::path::Path;
+
+    use sawtooth_sdk::signing::PrivateKey;
+    use serial_test::serial;
+    use tempfile::{tempdir, NamedTempFile};
+
+    const MOCK_PRIV_KEY_HEX: &str =
+        "d31e395bed0d9b2277b25d57523063d7d6b9db802d80549bc1362875cdcb83c6";
+
+    /// Verify that an error is retruned when the key file is not found.
```

Comment:
retruned -> returned (repeated in the tests below)

Reply:
Fixed